money-api/src/main/java/com/comcast/money/api/SpanId.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2012-2015 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.comcast.money.api;
+
+import java.util.Random;
+import java.util.UUID;
+
+/**
+ * A unique identifier for a Span
+ */
+public class SpanId {
+
+    private static final Random rand = new Random();
+    private static final String STRING_FORMAT = "SpanId~%s~%s~%s";
+
+    private final String traceId;
+    private final Long parentId;
```

Comment:
Seems like as-is, `parentId` and `selfId` will be null if we use either of the first two constructors. Which makes sense, I guess, since sometimes there isn't a parent span. Right? Also I'm not 100% sure what a selfId is used for. We should maybe add documentation to make the purpose of each of these clear.